ZENKO-2514 setup yarn install in worker container for caching

tcarmet opened the pull request:
# Introduction
This is the first PR and the step 1 to improve the build time in backbeat's CI.

I'll be splitting up the patches in multiple PR as some cannot be done in the lower versions of backbeat. And I want to keep a maximum number of improvements in the lower versions.

## Description
We are setting up the yarn install in the CI worker container so that we can
benefit some caching and improve the build speed.

tcarmet committed:
ZENKO-2514 setup yarn install in worker container for caching

Setting up the yarn install in the CI worker container so that we can
benefit some caching and improve the build speed.

bert-e commented:
# Hello tcarmet,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/backbeat/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/feature/ZENKO-2514-cache-yarn-install` with contents from `w/7.7/feature/ZENKO-2514-cache-yarn-install`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/feature/ZENKO-2514-cache-yarn-install origin/development/8.1
 $ git merge origin/w/7.7/feature/ZENKO-2514-cache-yarn-install
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/ZENKO-2514-cache-yarn-install
```

tcarmet committed:
ZENKO-2514 ensure we're not writing data in the container overlay

The CI when it starts running ShellCommands it will write some data like
the git repository. We need to ensure they all been written in the proper
path/volume so that we can benefit some good performances.

https://docs.docker.com/storage/volumes/

tcarmet commented:
/wait

bert-e commented:
# History mismatch



Merge commit #ff8e79609f09d9b7e5ce117ef59d13e8bc3cf305 on the integration branch
`w/7.6/feature/ZENKO-2514-cache-yarn-install` is merging a branch which is neither the current
branch `feature/ZENKO-2514-cache-yarn-install` nor the development branch
`development/7.6`.

It is likely due to a rebase of the branch `feature/ZENKO-2514-cache-yarn-install` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

tcarmet commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/feature/ZENKO-2514-cache-yarn-install` with contents from `w/7.7/feature/ZENKO-2514-cache-yarn-install`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/feature/ZENKO-2514-cache-yarn-install origin/development/8.1
 $ git merge origin/w/7.7/feature/ZENKO-2514-cache-yarn-install
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/ZENKO-2514-cache-yarn-install
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/ZENKO-2514-cache-yarn-install` into
`development/7.4`
* [w/7.6/feature/ZENKO-2514-cache-yarn-install](https://github.com/scality/backbeat/compare/development/7.6...w/7.6/feature/ZENKO-2514-cache-yarn-install) will be merged into `development/7.6` (pull request #819)
* [w/7.7/feature/ZENKO-2514-cache-yarn-install](https://github.com/scality/backbeat/compare/development/7.7...w/7.7/feature/ZENKO-2514-cache-yarn-install) will be merged into `development/7.7` (pull request #820)
* [w/8.1/feature/ZENKO-2514-cache-yarn-install](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/feature/ZENKO-2514-cache-yarn-install) will be merged into `development/8.1` (pull request #821)
* [w/8.2/feature/ZENKO-2514-cache-yarn-install](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/feature/ZENKO-2514-cache-yarn-install) will be merged into `development/8.2` (pull request #822)



The following branches will **NOT** be impacted:

* `stabilization/7.4.5`
* `stabilization/7.6.0`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @jonathan-gramain

* @rahulreddy

* @vrancurel

tcarmet committed:
ZENKO-2514 setup NODE_ENV as development on CI worker

bert-e commented:
# History mismatch



Merge commit #d774b32a36cab889dd1b86662f7754c8b12c26cf on the integration branch
`w/8.2/feature/ZENKO-2514-cache-yarn-install` is merging a branch which is neither the current
branch `feature/ZENKO-2514-cache-yarn-install` nor the development branch
`development/8.2`.

It is likely due to a rebase of the branch `feature/ZENKO-2514-cache-yarn-install` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/ZENKO-2514-cache-yarn-install` into
`development/7.4`
* [w/7.6/feature/ZENKO-2514-cache-yarn-install](https://github.com/scality/backbeat/compare/development/7.6...w/7.6/feature/ZENKO-2514-cache-yarn-install) will be merged into `development/7.6` (pull request #819)
* [w/7.7/feature/ZENKO-2514-cache-yarn-install](https://github.com/scality/backbeat/compare/development/7.7...w/7.7/feature/ZENKO-2514-cache-yarn-install) will be merged into `development/7.7` (pull request #820)
* [w/8.1/feature/ZENKO-2514-cache-yarn-install](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/feature/ZENKO-2514-cache-yarn-install) will be merged into `development/8.1` (pull request #821)
* [w/8.2/feature/ZENKO-2514-cache-yarn-install](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/feature/ZENKO-2514-cache-yarn-install) will be merged into `development/8.2` (pull request #823)



The following branches will **NOT** be impacted:

* `stabilization/7.4.5`
* `stabilization/7.6.0`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @jonathan-gramain

* @rahulreddy

* @vrancurel

tcarmet commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.6`

* :heavy_check_mark: `development/7.7`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches will **NOT** be impacted:

* `stabilization/7.4.5`
* `stabilization/7.6.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.6`

* :heavy_check_mark: `development/7.7`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `stabilization/7.4.5`
* `stabilization/7.6.0`



Please check the status of the associated issue ZENKO-2514.

Goodbye tcarmet.
